torchaudio/datasets/librilight_limited.py

```diff
@@ -0,0 +1,111 @@
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    extract_archive,
+)
+
+
+_URL = "https://dl.fbaipublicfiles.com/librilight/data/librispeech_finetuning.tgz"
+_CHECKSUM = "5d1efdc777b548194d7e09ba89126e2188026df9fd57aa57eb14408d2b2342af"
+
+
+def _get_files(path, subset, _ext_audio):
+    if subset == "10min":
+        files = sorted(str(p) for p in Path(path).glob("1h/0/*/*/*/*" + _ext_audio))
+    elif subset in ["1h", "10h"]:
+        files = [str(p) for p in Path(path).glob("1h/*/*/*/*/*" + _ext_audio)]
+        if subset == "10h":
+            files += [str(p) for p in Path(path).glob("9h/*/*/*/*" + _ext_audio)]
+        files = sorted(files)
+    else:
+        raise ValueError(f"Unsupported subset value. Found {subset}.")
+    return files
+
+
+def _load_item(file_path: str, ext_audio: str, ext_txt: str) -> Tuple[Tensor, int, str, int, int, int]:
+    fileid = os.path.basename(file_path)
+    path = os.path.dirname(file_path)
+    speaker_id, chapter_id, utterance_id = fileid.replace(ext_audio, "").split("-")
+
+    file_text = speaker_id + "-" + chapter_id + ext_txt
+    file_text = os.path.join(path, file_text)
+
+    fileid_audio = speaker_id + "-" + chapter_id + "-" + utterance_id
+    file_audio = fileid_audio + ext_audio
+    file_audio = os.path.join(path, file_audio)
+
+    # Load audio
+    waveform, sample_rate = torchaudio.load(file_audio)
+
+    # Load text
+    with open(file_text) as ft:
+        for line in ft:
+            fileid_text, transcript = line.strip().split(" ", 1)
+            if fileid_audio == fileid_text:
+                break
+        else:
+            # Translation not found
+            raise FileNotFoundError("Translation not found for " + fileid_audio)
+
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        int(speaker_id),
+        int(chapter_id),
+        int(utterance_id),
+    )
+
+
+class LibriLightLimited(Dataset):
+    """Create a Dataset for LibriLightLimited, which is the supervised subset of
+        LibriLight dataset.
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        subset (str, optional): The subset to use. Options: [``10min`, ``1h``, ``10h``]
+            (Default: ``10min``).
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+    """
+
+    _ext_txt = ".trans.txt"
+    _ext_audio = ".flac"
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        subset: str = "10min",
+        download: bool = False,
+    ) -> None:
+        assert subset in ["10min", "1h", "10h"], "`subset` must be one of ['10min', '1h', '10h']"
+
+        root = os.fspath(root)
+        self._path = os.path.join(root, "librispeech_finetuning")
+        archive = os.path.join(root, "librispeech_finetuning" + ".tgz")
+        if download:
```

Comment:
could you throw a runtime error if the dataset can not be found locally and download = False?

Reply:
@nateanl hmm ok, so currently this is what happens for quesst14:
-  if neither the folder or archive (.tgz) for the dataset exist, it will either throw a download error or correctly download and extract the files, which is the intended behavior
- if the folder doesn't exist but the archive exists, it will extract the archive files and therefore create the folder

for the second case, you're saying the archive file may not be correct and we extract incorrect files. To get around this, we could raise the download error if `self._path` does not exist, which means the user would download and extract the files themselves, then pass in the root corresponding to it. should we expect the users to extract the files themselves, and take the approach above?

I do think this is an issue we can't really predict, since even if they do have a folder with the correct name, we have no way of telling if the contents of the folder are expected without performing added checks for folder structure/files.